File: backtracking/combinations.py
def combine(n: int, k: int) -> list[list[int]]:
    def backtrack(curr: list[int], i: int):
        if len(curr) == k:
            ans.append(curr[:])
            return

        for num in range(i, n + 1):
            curr.append(num)
            backtrack(curr, num + 1)
            curr.pop()

    ans = []
    backtrack([], 1)
    return ans

File: backtracking/combinations.js
/**
 * @param {number} n
 * @param {number} k
 * @returns {number[][]}
 */
var combine = function(n, k) {
    const backtrack = (curr, i) => {
        if (curr.length == k)
        {
            ans.push([...curr]);
            return;
        }

        for (let num = i; num <= n; num++)
        {
            curr.push(num);
            backtrack(curr, num + 1);
            curr.pop();
        }
    }

    let ans  = [];
    backtrack([], 1);
    return ans;
}